objcgen/operatoroverloads.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using IKVM.Reflection;
+using Embeddinator;
+
+namespace ObjC
+{
+	public static class OperatorOverloads
+	{
+		struct OperatorInfo
+		{
+			public string MetadataName { get; set; }
+			public string FriendlyName { get; set; }
+			public int ArgumentCount { get; set; }
+
+			public OperatorInfo (string metadataName, string friendlyName, int argumentCount)
+			{
+				MetadataName = metadataName;
+				FriendlyName = friendlyName;
+				ArgumentCount = argumentCount;
+			}
+
+			public bool Contains (string name) => MetadataName == name || FriendlyName == name;
+		}
+
+		// From https://msdn.microsoft.com/en-us/library/ms229032(v=vs.110).aspx and https://msdn.microsoft.com/en-us/library/8edha89s.aspx
+		// with ambigious elements and non-overridable elements removed
+		static List<OperatorInfo> OperatorMapping = new List<OperatorInfo>() { 
+			new OperatorInfo ("op_Addition", "Add", 2),
+			new OperatorInfo ("op_Subtraction", "Subtract", 2),
+			new OperatorInfo ("op_Multiply", "Multiply", 2),
+			new OperatorInfo ("op_Division", "Divide", 2),
+			new OperatorInfo ("op_ExclusiveOr", "Xor", 2),
+			new OperatorInfo ("op_BitwiseAnd", "BitwiseAnd", 2),
+			new OperatorInfo ("op_BitwiseOr", "BitwiseOr", 2),
+			new OperatorInfo ("op_LogicalAnd", "And", 2),
+			new OperatorInfo ("op_LogicalOr", "Or", 2),
+			new OperatorInfo ("op_Assign", "Assign", 2),
+			new OperatorInfo ("op_LeftShift", "LeftShift", 2),
+			new OperatorInfo ("op_RightShift", "RightShift", 2),
+			new OperatorInfo ("op_SignedRightShift", "SignedRightShift", 2),
+			new OperatorInfo ("op_UnsignedRightShift  ", "UnsignedRightShift", 2),
```

Comment:
So my approach was:

- Start with list from: https://msdn.microsoft.com/en-us/library/ms229032(v=vs.110).aspx
- Remove all ones with multiple "friendly" names or multiple ones target the same "friendly" name (how can we tell without guessing).
- Then go look at https://msdn.microsoft.com/en-us/library/8edha89s.aspx and union that list with the ones I believe are overridable in C#.

I'm likely to have made mistakes.